src/main/kotlin/it/pagopa/wallet/services/WalletService.kt

```diff
@@ -176,6 +195,152 @@ class WalletService(
             }
     }
 
+    fun validateWallet(
+        orderId: UUID,
+        walletId: UUID
+    ): Mono<Pair<WalletVerifyRequestsResponseDto, LoggedAction<Wallet>>> {
+        val correlationId = UUID.randomUUID()
+        return mono { npgSessionRedisTemplate.findById(orderId.toString()) }
+            .switchIfEmpty { Mono.error(SessionNotFoundException(orderId)) }
+            .flatMap { session ->
+                walletRepository
+                    .findById(walletId.toString())
+                    .switchIfEmpty { Mono.error(WalletNotFoundException(WalletId(walletId))) }
+                    .filter { session.walletId == it.id }
+                    .switchIfEmpty {
+                        Mono.error(
+                            WalletSessionMismatchException(session.sessionId, WalletId(walletId))
+                        )
+                    }
+                    .filter { it.status == WalletStatusDto.INITIALIZED.value }
+                    .switchIfEmpty { Mono.error(WalletConflictStatusException(WalletId(walletId))) }
+                    .flatMap { wallet ->
+                        ecommercePaymentMethodsClient
+                            .getPaymentMethodById(wallet.paymentMethodId)
+                            .flatMap {
+                                when (it.paymentTypeCode) {
+                                    "CP" ->
+                                        confirmPaymentCard(
+                                            session.sessionId,
+                                            correlationId,
+                                            orderId,
+                                            wallet.toDomain()
+                                        )
+                                    else ->
```

Comment:
`confirmPayment` is only for CARDS / CP, in general the sessions are only used for the methods for which we need to get some data (such as the bin for CARDS / CP). For now the only method that runs a session are cards

Reply:
added new exception for this case